internal/buffer/buffer.go

```diff
@@ -1008,7 +1008,7 @@ func (b *Buffer) FindMatchingBrace(braceType [2]rune, start Loc) (Loc, bool, boo
 		leftChar = curLine[start.X-1]
 	}
 	var i int
-	if startChar == braceType[0] || leftChar == braceType[0] {
+	if startChar == braceType[0] || (leftChar == braceType[0] && startChar != braceType[1]) {
```

Comment:
Regarding Co-authored-by in the commit message: this is the only line I wrote, the rest of this patch was written by JoeKar.

BTW if you want to correct the commit message, I believe it requires force-pushing, so it might be also an occasion to clean up the history of patches in this PR (to ease its further review by @zyedidia or others), for example:
- Squash small uninteresting changes like fixing whitespaces and typos
- Squash "Fix cursor color being double-inverted" and "Fix cursor color being double-inverted, again" into one patch
- Remove both "Fix cursor color being double-inverted when braces are next to each other" and "Restore the simple logic in displayBuffer()" (the latter one just reverts the former one, and the former one is rather completely broken anyway)
- In this "Correct FindMatchingBrace() counting" patch, clarify in the commit message that it makes "match brace under the cursor" have priority over "match brace left of the cursor" in ambiguous cases